static/app/components/confirm.tsx

```diff
@@ -97,6 +103,10 @@ export type OpenConfirmOptions = {
    * Callback when user confirms
    */
   onConfirm?: () => void;
+  /**
+   * Prevents the modal from closing until the promise is resolved
+   */
+  onConfirmAsync?: () => Promise<unknown>;
```

Comment:
Can we actually just replace `onConfirm` with 

```
onConfirm?: () => (Promise<unknown> | void)
```

Reply:
I avoided that because there could be usages where an async function is passed but they don't want it to wait. Though that may not be an actual issue and would simplify the props

Reply:
In those cases can't you just

```tsx
onConfirm={() => {
  callmyFunction()
})
```

instead of `onConfirm={callmyFunction}`

Reply:
Yeah that's true. The only downside is that isn't not as explicit, but I think that's okay. Will make this change